source/application/TrustedModule.cpp

```diff
@@ -0,0 +1,92 @@
+﻿/*
+ * Copyright (c) 2017- Hourglass Resurrection Team
+ * Hourglass Resurrection is licensed under GPL v2.
+ * Refer to the file COPYING.txt in the project root.
+ */
+
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+
+#include <algorithm>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "Config.h"
+
+#include "TrustedModule.h"
+
+namespace
+{
+    std::map<DWORD, std::vector<std::wstring>> s_trusted_modules;
+
+    UINT CountSharedPrefixLength(const std::wstring& a, const std::wstring& b)
+    {
+        size_t shortest_path = std::min(a.size(), b.size());
+        UINT i;
+        for (i = 0; i < shortest_path && tolower(a[i]) == tolower(b[i]); ++i);
+        return i;
+    }
+    UINT CountBackslashes(const std::wstring& str)
+    {
+        return std::count(str.begin(), str.end(), L'\\');
+    }
+
+}
+
+/*
+ * Hacky... yucky
+ * -- Warepire
+ */
+extern std::wstring injected_dll_path;
+extern std::wstring sub_exe_filename;
+
+namespace TrustedModule
+{
+    void OnLoad(DWORD process_id, const std::wstring& path)
+    {
+        /*
+        * If we can "trust" a newly loaded module, add it to the list.
+        */
+        UINT proximity = 2;
+        if (!wcsicmp(path.c_str(), injected_dll_path.c_str()))

```

Comment:
It's `wcsicmp`, case insensitive. I don't see why it needs to be that now though. I will change it.